lib/vanagon/project.rb

```diff
@@ -692,5 +718,38 @@ def load_upstream_settings(upstream_project_name, upstream_git_url, upstream_git
         upstream_project.cleanup
       end
     end
+
+    # Load the settings hash for the current project/platform combination from a
+    # yaml file as produced by `publish_yaml_settings`. file:// and http:// URIs
+    # are accepted. If the URI uses http://, a sha1 URI is also required.
+    #
+    # @param settings_uri [String] A URI to a yaml settings file
+    # @param settings_sha1_uri [String] A URI to a sha1sum file for the yaml settings file
+    # @raise [Vanagon::Error] when the settings file can't be found
+    def load_yaml_settings(settings_uri, settings_sha1_uri = nil) # rubocop:disable Metrics/AbcSize
+      source_type = Vanagon::Component::Source.determine_source_type(settings_uri)
+
+      if %i[unknown git].include?(source_type)
+        message = if settings_uri =~ /^file/
```

Comment:
[nitpick] I find this way of assigning variables hard to read, would prefer either
```
message = "Can't inherit settings from '#{settings_uri}'. Only http and file URIs are valid."
if settings_uri =~ /^file/
  message = "Tried to load YAML settings from '#{settings_uri}', but the file doesn't exist."
end
```

or

```
if settings_uri =~ /^file/
  message = "Tried to load YAML settings from '#{settings_uri}', but the file doesn't exist."
else
  message = "Can't inherit settings from '#{settings_uri}'. Only http and file URIs are valid."
end
```

Reply:
:+1: I like those styles better, too -- updated